src/Orleans.Core/Messaging/GatewayManager.cs

```diff
@@ -274,6 +279,8 @@ private void UpdateLiveGatewaysSnapshot(IEnumerable<Uri> currentKnownGateways, T
                             prevRefresh);
                 }
             }
+
+            this.messageCenter.CleanupGatewayConnections(cachedLiveGateways);
```

Comment:
Given that `null` is being passed in for `messageCenter` in the `GatewaySelectionTest.cs` file, should this be a `?.` call instead?